src/canvas/PatternPath.tsx

```diff
@@ -1,16 +1,13 @@
 import { Point } from './Point';
 import { PatternPathType, ToolType } from './Enums';
-import { CurveFitter } from './CurveFitter';
-import { Curve } from './Curve';
-
-export class PatternPath implements IPatternPath {
-    private _points: Point[];
-    private _type: PatternPathType;
-    private _toolType: ToolType;
-    private _path2D: Path2D;
-    private _isPath2DValid: boolean;
-    private _lastIndexAddedToPath2D: number;
-    private _fittedCurve: Curve | null;
+
+export abstract class PatternPath implements IPatternPath {
+    protected _points: Point[];
+    protected _type: PatternPathType;
+    protected _toolType: ToolType;
```

Comment:
Why keep the tooltype data? I don't see it used anywhere. I'd take that out of the classes.

Reply:
That is a good point, I will remove it. My reason for leaving it was that PatternPaths still have a tool type. However, I think you are right that we don't need to codify it in a member variable anymore.